### ISSUE TITLE
Update bookshelf using material design

### ISSUE THREAD

**Issue by giovariot** [SufficientlySecure/document-viewer#142]
I think the wooden wallpaper in library should be removed. Would also be an easy step toward material guidelines :)

**Comment by tuxor1337** [comment]
I second this. I would also prefer a cleaner UI design (that includes new/simpler icons etc.).

**Comment by KrasnayaPloshchad** [comment]
I found DV has another interface doesn’t use wooden wallpapers, which should enabled in the settings. I think this interface can be modified to identical material guidelines.

**Comment by ericwa** [comment]
Ah you're right, disabling "All Settings -> File Browser -> Use bookcase view" gives a list view instead of the bookcase.
But I think it's still worth getting rid of the ugly wood texture anyway.

**Comment by KrasnayaPloshchad** [comment]
I think the wooden wallpaper can be keeped, because this including a easter egg.

**Comment by hackerb9** [comment]
What? I like the wooden bookcase view. I just wish I could create my own bookcases and rearrange the books so that I can access them by spatial memory instead of searching.

**Comment by ericwa** [comment]
I dropped the wood wallpaper for now, here's what it currently looks like:
IMHO it's cleaner, maybe too dark. Could use wood colours maybe? We don't have an app colour scheme currently.

![screenshot_2016-09-30-22-08-25-930_org sufficientlysecure viewer](https://cloud.githubusercontent.com/assets/239161/19011793/b9fa5b2c-875e-11e6-94d5-d41e5c021b86.png)

**Comment by KrasnayaPloshchad** [comment]
After you dropped the wood wallpaper, you should remove the outline of thumbnails, replace it by shadow. BTW I think the wooden interface should be moved as alternate UI...

**Comment by hackerb9** [comment]
I agree with Krasnaya. Please keep the wooden interface as an option instead of completely dropping it. Thanks.

**Comment by ericwa** [comment]
Ok, I'll restore it as an option.

I'd prefer to keep the new scrolling tabs I added though, what do you think of this screenshot?  https://cloud.githubusercontent.com/assets/239161/18811338/bcf3f54e-8269-11e6-8307-445d96094632.png 

IMO the scrolling tabs are a usability improvement over the old header with left/right buttons because you can see the neighbouring bookshelves.

re: the material design bookshelf, yeah, there is more work to do.
It should either use the [cards](https://material.google.com/components/cards.html#cards-usage) or [grid list](https://material.google.com/components/grid-lists.html#grid-lists-behavior) pattern, not sure which.

**Comment by hackerb9** [comment]
Thanks for restoring the wooden bookcase. I think the tabs are an improvement over the uninformative left/right arrows. (One can presumably still just swipe left and right anyhow, so the arrows are also functionally unnecessary).

**Comment by hackerb9** [comment]
In regards to grid list versus cards, can I presume you mean a card collection? If so, the material guidelines has a handy checklist:

"Use a card layout when displaying content that:

    (1) As a collection, comprises multiple data types, such as images, movies, and text
 
   (2) Does not require direct comparison (a user is not directly comparing images or text)

    (3) Supports content of highly variable length, such as comments

    (4) Contains interactive content, such as +1 buttons or comments, or

    (5) Would otherwise be in a grid list but needs to display more content to supplement the image"

Of those considerations, I think probably #4 and #5 are most salient, but they really depends upon how you imagine people interacting with the book shelf. 

Personally, I like the small grid like option currently in use as I can see more titles than a card collection. The only extra interactions I would want with the tiles would to be able to drag them to different positions, shelves, or trash.

**Comment by dschuermann** [comment]
Just for reference how Google does it in Google Books:
![](https://lh5.ggpht.com/j29rCi5xMY-IwTWjuomZKEiyGLvWwsN86S7sp_RbpY5Y8yo_rYPMokcKn2KDi558eQoY=h900)

**Comment by KrasnayaPloshchad** [comment]
Recently I found two libraries can be used to make some redesigns.

https://github.com/HeinrichReimer/material-drawer
https://github.com/afollestad/material-dialogs

**Comment by dschuermann** [comment]
As a drawer I can recommend https://github.com/mikepenz/MaterialDrawer . We use it at Openkeychain and its updated regularly. For other things, http://www.android-arsenal.com is a great resource.

**Comment by KrasnayaPloshchad** [comment]
> As a drawer I can recommend https://github.com/mikepenz/MaterialDrawer . We use it at Openkeychain and its updated regularly. For other things, http://www.android-arsenal.com is a great resource.

OK this si also a good library to get the Material design.
